api/controllers/v1/sponsorship.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"net/http"
+	"strconv"
+
+	database "github.com/GDGVIT/gibspons-backend/database/dbops"
+	"github.com/GDGVIT/gibspons-backend/database/tables"
+	"github.com/GDGVIT/gibspons-backend/utility"
+	"github.com/gin-gonic/gin"
+)
+
+func CreateSponsorship(c *gin.Context) {
+	var body struct {
+		PocID       int
+		Email       string
+		Description string
+		Status      string
+	}
+
+	if err := c.Bind(&body); err != nil {
+		utility.GinCtxError(c, "Failed to create sponsorship, check body")
+		return
+	}
+
+	spon := tables.Sponsorships{
+		PocID:       body.PocID,
+		Email:       body.Email,
+		Description: body.Description,
+		Status:      body.Status,
+	}
+
+	if err := database.CreateSponsorship(&spon); err != nil {
+		utility.GinCtxError(c, "Failed to create sponsorship, check db logs")
+		return
+	}
+
+	c.JSON(http.StatusCreated, gin.H{
+		"sponsorship": spon,
+	})
+}
+
+func GetSponsorshipById(c *gin.Context) {
+	id := c.Param("id")
+	idInt, _ := strconv.Atoi(id)
+	spon, err := database.GetSponsorshipById(idInt)
+	if err != nil {
+		utility.GinCtxError(c, err.Error())
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{
+		"sponsorship": spon,
+	})
+}
```

Comment:
Have a separate Controller and service layer

Reply:
Elaborate please